apps/resource_manager/lib/credentials/schemas/password.ex

```diff
@@ -12,48 +12,100 @@ defmodule ResourceManager.Credentials.Schemas.Password do
 
   import Ecto.Changeset
 
+  alias ResourceManager.Credentials.Commands.PasswordIsAllowed
   alias ResourceManager.Identities.Schemas.User
 
   @typedoc """
   Abstract password module type.
   """
   @type t :: %__MODULE__{
-          id: binary(),
-          user: User.t(),
-          algorithm: String.t(),
+          id: Ecto.UUID.t(),
+          user: User.t() | Ecto.Association.NotLoaded.t(),
+          user_id: Ecto.UUID.t(),
+          value: String.t() | nil,
           password_hash: String.t(),
-          inserted_at: NaiveDateTime.t(),
-          updated_at: NaiveDateTime.t()
+          algorithm: String.t(),
+          salt: integer(),
+          inserted_at: Datetime.t(),
+          updated_at: Datetime.t()
         }
 
-  @possible_algorithms ~w(argon2 bcrypt pbkdf2)
+  # Default module arguments
+  @default_algorithm "argon2"
+  @default_salt 16
+
+  # Changeset validation arguments
+  @acceptable_algorithms ~w(argon2 bcrypt pbkdf2)
 
   @required_fields [:password_hash]
-  @foreign_key_fields [:user_id]
-  @optional_fields [:algorithm]
+  @optional_fields [:user_id, :value, :algorithm, :salt]
   schema "passwords" do
```

Comment:
if you set :pasword, :string, virtual: true here, you dont need to validate if password is binary in the below funcitons, right?